code/modules/atmospherics/environmental/LINDA_system.dm

```diff
@@ -8,42 +10,47 @@
 		else
 			return CanAtmosPass
 
-/turf/closed/CanAtmosPass = ATMOS_PASS_NO
+/turf/CanAtmosPass = ATMOS_PASS_NO
+/turf/CanAtmosPassVertical = ATMOS_PASS_NO
 
 /turf/open/CanAtmosPass = ATMOS_PASS_PROC
-/turf/open/CanAtmosPass(turf/T)
-	var/R
+/turf/open/CanAtmosPassVertical = ATMOS_PASS_PROC
+
+/turf/open/CanAtmosPass(turf/T, vertical = FALSE)
+	var/dir = vertical? get_dir_multiz(src, T) : get_dir(src, T)
+	var/opp = dir_inverse_multiz(dir)
+	var/R = FALSE
+	if(vertical && !(zAirOut(dir, T) && T.zAirIn(dir, src)))
+		R = TRUE
 	if(blocks_air || T.blocks_air)
-		R = 1
-
+		R = TRUE
+	if (T == src)
+		return !R
 	for(var/obj/O in contents+T.contents)
 		var/turf/other = (O.loc == src ? T : src)
-		if(!CANATMOSPASS(O, other))
-			R = 1
+		if(!(vertical? (CANVERTICALATMOSPASS(O, other)) : (CANATMOSPASS(O, other))))
+			R = TRUE
 			if(O.BlockSuperconductivity()) 	//the direction and open/closed are already checked on CanAtmosPass() so there are no arguments
-				var/D = get_dir(src, T)
-				atmos_supeconductivity |= D
-				D = get_dir(T, src)
-				T.atmos_supeconductivity |= D
-				return 0						//no need to keep going, we got all we asked
+				atmos_supeconductivity |= dir
+				T.atmos_supeconductivity |= opp
+				return FALSE						//no need to keep going, we got all we asked
 
-	atmos_supeconductivity &= ~get_dir(src, T)
-	T.atmos_supeconductivity &= ~get_dir(T, src)
+	atmos_supeconductivity &= ~dir
+	T.atmos_supeconductivity &= ~opp
 
 	return !R
 
-
-
 /atom/movable/proc/BlockSuperconductivity() // objects that block air and don't let superconductivity act. Only firelocks atm.
-	return 0
+	return FALSE
 
 /turf/proc/CalculateAdjacentTurfs()
-	var/list/atmos_adjacent_turfs = src.atmos_adjacent_turfs
-	for(var/direction in GLOB.cardinals)
-		var/turf/T = get_step(src, direction)
-		if(!T)
+	var/canpass = CANATMOSPASS(src, src)
```

Comment:
These two vars are only used once in this proc. You can safely squash them into the places where they're used.

Reply:
These two variables are actually encapsulated in a `for(var/direction in GLOB.cardinals_multiz)` loop. Doing what you suggest here would be an obvious performance drawback.

Reply:
ohhhhhhhh. yeah im dum this is good